### PR TITLE
Show disassembly var tooltips

### PR DESCRIPTION
**Pull request by frmdstryr**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

This adds tooltips when hovering over a register or variable to the disassembly and disassembly graph views as requested in https://github.com/rizinorg/cutter/issues/2532

**Test plan (required)**

Enable the setting "Show known variable values when hovering" in the preferences. Start a debug or emulation session and hover over various registers / variables and check the tooltip.

**Closing issues**

Does not implement everything in https://github.com/rizinorg/cutter/issues/2532 but it is a start. 

Any feedback/suggestions on improving the tooltips is welcome. Thank you!

### CODE REVIEW

File: src/common/DisassemblyPreview.cpp [hunk 1]
```diff
@@ -82,3 +82,74 @@ RVA DisassemblyPreview::readDisassemblyOffset(QTextCursor tc)
...
+                    return true;
+                }
+            }
+        }
+
+        if (offset != RVA_INVALID) {
+            auto vars = Core()->getVariables(offset);
+            for (auto &var : vars) {
+                if (var.name == selectedText) {
+                    auto msg = Core()->cmdRawAt(QString("afvd %1").arg(var.name), offset);
```
Comment (XVilka) [inline src/common/DisassemblyPreview.cpp]: Please don't use commands. We removed them in the favor of the API:  https://github.com/rizinorg/cutter/issues/2666

File: src/common/DisassemblyPreview.cpp [hunk 2]
```diff
@@ -82,3 +82,74 @@ RVA DisassemblyPreview::readDisassemblyOffset(QTextCursor tc)
...
+                auto core = Core()->core();
+                RzPlatformTarget *arch_target = core->analysis->arch_target;
+                if (arch_target && arch_target->profile) {
+                    ht_up_foreach(arch_target->profile->registers_mmio, lookup_mmio_addr_cb, &ctx);
+                }
+            }
+            if (ctx.mmio_address != RVA_INVALID && offset != RVA_INVALID) {
+                auto addr = QString::number(ctx.mmio_address, 16);
+                QString size = "8"; // TODO: Determine proper size
+                auto value = Core()->cmdRawAt(QString("px %1 @ 0x%2").arg(size, addr), offset);
```
Comment (XVilka) [inline src/common/DisassemblyPreview.cpp]: Same

File: src/common/Configuration.cpp [hunk 3]
```diff
@@ -134,6 +134,7 @@ static const QHash<QString, QVariant> asmOptions = { { "asm.esil", false },
                                                      { "asm.flags.real", true },
                                                      { "asm.reloff", false },
                                                      { "asm.reloff.flags", false },
+                                                     { "asm.varTooltips", false },
```
Comment (XVilka) [inline src/common/Configuration.cpp]: These are configuration variables that exist also in Rizin, it should be controlled differently

**Review by XVilka — CHANGES_REQUESTED** [review]

File: src/common/DisassemblyPreview.cpp (hunk 1 above)
Comment (frmdstryr) [reply src/common/DisassemblyPreview.cpp]: It looks like `rz_core_analysis_var_display` is in the private api or is there another way to get the value?

Comment (wargio) [reply src/common/DisassemblyPreview.cpp]: just do this:
```c
RzStrBuf *sb = rz_strbuf_new(NULL);
RzAnalysisFunction *fcn = analysis_get_function_in(core->analysis, offset);
if (!fcn) {
	return rz_strbuf_drain(sb);
}
void **it;
rz_pvector_foreach (&fcn->vars, it) {
	RzAnalysisVar *p = *it;
	char *r = rz_core_analysis_var_display(core, p, add_name);
	rz_strbuf_append(sb, r);
	free(r);
}
return rz_strbuf_drain(sb);
```

Comment (XVilka) [reply src/common/DisassemblyPreview.cpp]: You can send a PR to Rizin changing the visibility and update the Rizin submodule. Probably also makes sense to rename it from `_display()` to `_as_string()`

Comment (wargio) [reply src/common/DisassemblyPreview.cpp]: i agree, and potentially you can modify the RZ_IPI into RZ_API, so you don't need to add this code to cutter.

**Comment by frmdstryr** [comment]
It seems to work fine for registers but stack variables do not seem to be correct. 

Screenshot_20230721_084938

The `rz_analysis_var_addr` used inside `rz_core_analysis_var_display` does not return the proper address. After a little more digging `rz_reg_getv` is returning 0 for rbp insead of the actual value, since the stackoff is negative it overflows to a garbage address. 

Any ideas as to what is going on? Edit: This is when debugging.

**Comment by frmdstryr** [comment]
Potentially fixed by https://github.com/rizinorg/rizin/compare/dev...frmdstryr:rizin:make-rz_core_analysis_var_display-work-when-debugging

**Comment by wargio** [comment]
please update also the rizin submodule

**Comment by XVilka** [comment]
@frmdstryr please update Rizin, and make it green

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@frmdstryr thank you!

**Pull request merged**